### PR TITLE
Added Bounty Providers Endpoint

### LINKED ISSUE #1589: Create new endpoint to filter for providers for super admin
### Context
Create endpoint to filter for providers for super admin. This is currently handled in frontend. Should be handled in backend.

For context:
https://github.com/stakwork/sphinx-tribes-frontend/pull/341#issuecomment-1979348241

### PR DESCRIPTION
**Pull request by elraphty**
## Describe your changes

This feature adds a bounty providers list endpoint.

## Issue ticket number and link

Closes #1589 

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] I have tested on Chrome and Firefox
- [ ] I have tested on a mobile device

### CODE REVIEW

**Comment by kevkevinpal** [comment]
In this PR we are adding a query param for providers
https://github.com/stakwork/sphinx-tribes-frontend/pull/341/files#diff-b99fbceed6dc4aa0085944eae64088cc235ec10e88d81c8e40a721fb1949624cR93

for the `/metrics/bounties` endpoint, we should only return metrics with that provider

https://github.com/stakwork/sphinx-tribes/blob/master/routes/metrics.go#L21

**Comment by kevkevinpal** [comment]
do you think we should fix the endpoint using the query param instead of creating a new endpoint?

**Comment by kevkevinpal** [comment]
ohh wait nvm disregard my previous comments it makes sense why we'd need this endpoint for it to work with the frontend now

**Comment by elraphty** [comment]
> ohh wait nvm disregard my previous comments it makes sense why we'd need this endpoint for it to work with the frontend now

Yeah, the provider's list for the bounties is just to select bounties for certain providers, but we need a paginated list of providers based on the date range and status selected on the provider's modal.

File: db/metrics.go [hunk 1]
```diff
@@ -190,7 +190,7 @@ func (db database) GetBountiesByDateRange(r PaymentDateRange, re *http.Request)
 	} else {
 		orderQuery = " ORDER BY " + sortBy + "" + "DESC"
 	}
-	if limit > 1 {
+	if limit > 0 {
```
Comment (kevkevinpal) [inline db/metrics.go:193]: why is this changed?

**Review by kevkevinpal — APPROVED** [review]
lgtm minus the one comment

File: db/metrics.go (hunk 1 above)
Comment (elraphty) [reply db/metrics.go:193]: The default limit is 1, so any number lesser than 1 should not trigger the limit query.
